### PR TITLE
[RFC][UN-15612] Add calculation for copy provisioning

### PR DESCRIPTION
**Pull request by hfcorreia**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+type Config struct {
+	readCapacityUnits  int64
+	writeCapacityUnits int64
+	readWorkers        int
+	writeWorkers       int
+}
+
+func NewConfig(readUnits, writeUnits, readWorkers, writeWorkers int) Config {
+	return Config{
```
Comment (jkrnak) [inline config.go]: Will need to support on-demand mode which doesn't have provisioned read/write units.

Comment (hfcorreia) [reply config.go]: This config is just to store the parameter provided from the command line. We check for the type of billing when we fetch the current provisioning of the table, if the billing mode is `PER_REQUEST` we don't set the provisioning capacity for the table. The code then just skips the provisioning update.

Comment (jkrnak) [reply config.go]: roger that

File: config.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		readCapacityUnits:  int64(readUnits),
+		writeCapacityUnits: int64(writeUnits),
+		readWorkers:        readWorkers,
+		writeWorkers:       writeWorkers,
+	}
+}
+
+func (c Config) Provisioning(current Provisioning) Provisioning {
+	src := current.Source
+	if current.Source != nil && c.readCapacityUnits > current.Source.Read {
```
Comment (jmdalmeida) [inline config.go]:
```suggestion
	if src != nil && c.readCapacityUnits > src.Read {
```

File: config.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		src = &Capacity{Read: c.readCapacityUnits, Write: src.Write}
+	}
+
+	trg := current.Target
+	if trg != nil && c.writeCapacityUnits > current.Target.Write {
```
Comment (jmdalmeida) [inline config.go]:
```suggestion
	if trg != nil && c.writeCapacityUnits > trg.Write {
```

File: pkg/cmd/copytable/copytable_test.go [hunk 4]
```diff
@@ -13,154 +13,92 @@ import (
...
+					Copier:      copierMock,
+					Provisioner: provisionerMock,
+					Config:      testCase.config,
+				}
+
+				err := copytable.RunCopyTable(deps)
+
+				assert.Equal(t, testCase.expectedError, err)
+
+				copierMock.AssertExpectations(st)
```
Comment (jmdalmeida) [inline pkg/cmd/copytable/copytable_test.go]: missing `provisionerMock.AssertExpectations`

File: provisioner.go [hunk 5]
```diff
@@ -39,45 +39,51 @@ func (dc provisioningService) Update(provisioning Provisioning) (Provisioning, e
...
 
-func needsProvisioningUpdate(c1, c2 Capacity) bool {
-	return c1.Read != c2.Read || c1.Write != c2.Write
-}
-
-type Provisioning struct {
-	Source dynamodb.TableDescription
-	Target dynamodb.TableDescription
+func needsProvisioningUpdate(c1, c2 *Capacity) bool {
+	return (c1 != nil && c2 != nil) && (c1.Read != c2.Read || c1.Write != c2.Write)
```
Comment (jmdalmeida) [inline provisioner.go]:
```suggestion
	return c1 != nil && c2 != nil && (c1.Read != c2.Read || c1.Write != c2.Write)
```

**Pull request merged**